app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt

```diff
@@ -70,6 +78,38 @@ class MainActivity : AppCompatActivity(), ApplicationContract.View {
     private fun submitButtonTapped() {
         val originStation = findViewById<Spinner>(R.id.departureSpinner).selectedItem.toString()
         val finalStation = findViewById<Spinner>(R.id.arrivalSpinner).selectedItem.toString()
-        presenter.makeTrainSearch(originStation, finalStation)
+        val dateTime = getSelectedDate()
+        presenter.makeTrainSearch(originStation, finalStation, dateTime)
+
+
+    }
+    private fun getSelectedDate() : String {
+        val selectedDate = findViewById<DatePicker>(R.id.datePicker)
+        val day = addZeroToDateTime(selectedDate.dayOfMonth)
+        val month = addZeroToDateTime(selectedDate.month + 1)
+        val year = "${selectedDate.year}"
+
+        val selectedTime = findViewById<TimePicker>(R.id.timePicker)
+        val hour = addZeroToDateTime(selectedTime.hour)
+        val minute = addZeroToDateTime(selectedTime.minute)
+
+        return year + "-" + month + "-" + day + "T" + hour + ":" + minute + ":00.000+01:00"
+    }
```

Comment:
I'm not a big fan of this function, as we're doing a lot of time handling ourselves - I would expect there to be a function that allows you to just add a time to a date and it will give it to you as a DateTime (then you could use the DateTime -> String behaviour I explained in a previous comment to get it in the correct format to pass to the API)

Reply:
Could look at using the `Calendar` class, then something like
```
val calendar = Calendar.getInstance()
calendar.set(
     datePicker.year,
     datePicker.month,
     datePicker.dayOfMonth,
     timePicker.hours,
     timePicket.minutes
)
return calendar.time
```